src/config.rs

```diff
@@ -375,6 +376,24 @@ impl<'a> Cfg<'a> {
         self.toolchain_override = Some(toolchain_override.to_owned());
     }
 
+    pub(crate) fn set_auto_install(&mut self, mode: AutoInstallMode) -> Result<()> {
+        self.settings_file.with_mut(|s| {
+            s.auto_install = Some(mode);
+            Ok(())
+        })?;
+        (self.notify_handler)(Notification::SetAutoInstall(mode.as_str()));
+        Ok(())
+    }
+
+    pub(crate) fn should_auto_install(&self) -> Result<bool> {
+        if let Ok(mode) = self.process.var("RUSTUP_AUTO_INSTALL") {
+            Ok(mode != "0")
```

Comment:
Should I be smarter here? So if auto-install is disabled then it checks `mode != "1"` instead of `mode != "0"`. Basically to ensure it defaults to the setting if the environment variable is not recognised.

Reply:
On the other hand, that would be more awkward. I'm just worried about `RUSTUP_AUTO_INSTALL=false` having surprising behaviour if auto-install is disabled in the settings.

Reply:
@ChrisDenton I understand your concerns, idiomatically we should have used `str2bool`, yes. However since we're documenting `1` and `0` ONLY, I personally don't think that would be that much of a problem here. In the past, I believe, the convention is that, if a flag is enabled by default, the only way to disable it is to set it to a `0`.